### DISCUSSION TITLE
how to enable webworker in tauri

### DISCUSSION THREAD

**Discussion by MoYuan-CN** [tauri-apps/tauri#9595] [Q&A]
when i using monaco-editor and define a custom theme,
it tell me in console that failed to create webworker, like:
```log
simpleWorker.js:20 
Could not create web worker(s). Falling back to loading web worker code in main thread, which might cause UI freezes. Please see https://github.com/microsoft/monaco-editor#faq
```
in faq it says need to run website in `http` or `https`, but tauri already running on http mode.
so how can i use it?

(sorry for my poor english)

**Comment by FabianLars** [comment]
Can you post the output of the `tauri info` command please so we know what platform / version we're dealing with?

**Reply by MoYuan-CN** [reply]
here is it:

```
[✔] Environment
    - OS: Windows 10.0.26100 X64
    ✔ WebView2: 124.0.2478.67
    ✔ MSVC: Visual Studio ���ɹ��� 2022
    ✔ rustc: 1.77.1 (7cf61ebde 2024-03-27)
    ✔ cargo: 1.77.1 (e52e36006 2024-03-26)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.11.0
    - npm: 10.5.2

[-] Packages
    - tauri [RUST]: 1.6.2
    - tauri-build [RUST]: 1.5.1
    - wry [RUST]: 0.24.8
    - tao [RUST]: 0.16.9
    - @tauri-apps/api [NPM]: 1.5.4
    - @tauri-apps/cli [NPM]: 1.5.12

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```

platform Windows 11

**Reply by FabianLars** [reply]
Hmm, okay. I'm indeed not aware of any issues on Windows and the rest looks alright too 🤔 

Can you check if it works fine in Edge or alternatively any chromium based browser (edge is the closest to the webview we use)?

**Reply by MoYuan-CN** [reply]
no, it doesn't works in Edge.
is this problem cause by monaco-editor?

**Reply by FabianLars** [reply]
I think so, yes. But i am not 100% sure because i never worked with it myself.

**Reply by MoYuan-CN** [reply]
ok. wait for you message.

**Comment by MoYuan-CN** [comment]
umm…… i found something may can fix my problem
is tauri running in a sandbox mode? if yes, how to turn off it?

**Reply by FabianLars** [reply]
Hmm, not really. At least nothing else than the default browser sandbox every website runs in in normal browsers too.

**Reply by MoYuan-CN** [reply]
got it.

**Comment by MoYuan-CN** [comment]
let me ask it in monaco-editor repo: https://github.com/microsoft/monaco-editor/discussions/4486

**Comment by MoYuan-CN** [comment]
here is something news.
i trying to init a monaco editor in a simple vue project, problem still there.
this is an issue about monaco editor.

anyway, thanks for your help
